common/courses/states.ts

```diff
@@ -129,15 +131,41 @@ export async function editSubcourse(subcourse: Subcourse, update: Partial<Subcou
     if (!can.allowed) {
         throw new Error(`Cannot edit Subcourse(${subcourse.id}) reason: ${can.reason}`);
     }
+    const participantCount = await prisma.subcourse_participants_pupil.count({ where: { subcourseId: subcourse.id } });
 
     const isMaxParticipantsChanged: boolean = Boolean(update.maxParticipants);
+    const isGroupChatTypeChanged: boolean = Boolean(update.groupChatType);
```

Comment:
Even if it's not a mandatory field it could be that the groupChatType is present but hasn't changes. We should check if the last type equals the current value present in the update.